server/api/crud/model_monitoring/deployment.py

```diff
@@ -1203,6 +1212,54 @@ def set_credentials(
             ),
         )
 
+    def _is_the_same_cred(
+        self, endpoint_store_connection: str, stream_path: str, tsdb_connection: str
+    ) -> bool:
+        credentials_dict = {
+            key: server.api.crud.Secrets().get_project_secret(
+                project=self.project,
+                provider=mlrun.common.schemas.SecretProviderName.kubernetes,
+                secret_key=key,
+                allow_secrets_from_k8s=True,
+            )
+            for key in mlrun.common.schemas.model_monitoring.ProjectSecretKeys.mandatory_secrets()
+        }
+
+        old_store = credentials_dict[
+            mlrun.common.schemas.model_monitoring.ProjectSecretKeys.ENDPOINT_STORE_CONNECTION
+        ]
+        if old_store != endpoint_store_connection:
+            logger.debug(
+                "User provided different endpoint store connection",
+                old_store=old_store,
+                new_store=endpoint_store_connection,
+            )
```

Comment:
I think that just keeping the message without printing the actual creds is enough (same for the other logs)
```suggestion
        if old_store != endpoint_store_connection:
            logger.debug(
                "User provided different endpoint store connection",
            )
```